cl8/users/importers.py

```
@@ -751,3 +754,49 @@ def add_cat_responses_to_profiles():
         if join_req:
             prof.bio = join_req.bio_text_from_join_request()
             prof.save()
+
+def gsheet_join_request_for_email(email: str) -> CATJoinRequest:
+    """
+    Return the matching join request for the provided email address, from
+    the CAT joining form spreadsheet
+    """
+    gc = gspread.service_account(filename=settings.GSPREAD_SERVICE_ACCOUNT)
+    gsheet = gc.open_by_key(settings.GSPREAD_KEY)
+    responses_worksheet = gsheet.worksheet(CAT_RESPONSES_WORKSHEET)
+
+    matching_cell_for_email = responses_worksheet.find(email)
```

Should we handle the case where there are multiple rows with the same email? I could see that causing confusion, and maybe that should return an error.

---

Yes, this is a very good point - we do have cases where we do indeed have more match. In this scenario, I think returning the last match would make sense as it would be the most recent submission.